### PR TITLE
cardano-cli Plutus Datum JSON format support

### PR DESCRIPTION
**Pull request by rooooooooob**
Implements both JSON formats used by cardano-node/cardano-cli

Based on: https://github.com/input-output-hk/cardano-node/blob/c1efb2f97134c0607c982246a36e3da7266ac194/cardano-api/src/Cardano/Api/ScriptData.hs#L254

### CODE REVIEW

File: rust/src/plutus.rs [hunk 1]
```diff
@@ -616,7 +616,207 @@ impl Strings {
 
 
 
+// json
 
+#[wasm_bindgen]
```
Comment (rooooooooob) [inline rust/src/plutus.rs:621]: I'll document this better/fully once we implement the rest of the BasicConversions format

File: rust/src/utils.rs [hunk 2]
```diff
@@ -675,6 +675,7 @@ impl BigInt {
             return None;
         }
         match u64_digits.len() {
+            0 => Some(to_bignum(0)),
```
Comment (rooooooooob) [inline rust/src/utils.rs:678]: unrelated to this PR, this was just a bug similar to the issue we had for serializing 0 as `BigInt` before.

**Review by SebastienGllmt — APPROVED** [review]

**Comment by vsubhuman** [comment]
@rooooooooob , resolve conflicts please when possible

**Comment by vsubhuman** [comment]
@lisicky : review and resolve conflicts plz

**Review by lisicky — COMMENTED** [review]
/check

**Review by lisicky — APPROVED** [review]
To reduce risk of conflicts, we need merge https://github.com/Emurgo/cardano-serialization-lib/pull/345  before this PR

**Review by lisicky — COMMENTED** [review]
/check

**Review by lisicky — COMMENTED** [review]
/check

**Review by lisicky — COMMENTED** [review]
/check

**Review by lisicky — COMMENTED** [review]
/check

**Pull request merged**